scripts/helpers/lock-editing.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Given a block's client ID and an attribute key, locks post saving in Gutenberg.
+ * A lock is created under the name undefined-lock-[blockClientId]-[attributeKey], 
+ * meaning that a lock can be created for each attribute key of each block.
+ * Multiple locks can be present and all of them have to be removed before saving is unlocked.
+ * 
+ * @param {string} blockClientId A block's client ID
+ * @param {string} attributeKey The attribute key
+ */
+export const lockPostEditing = (blockClientId, attributeKey) => {
+  wp.data.dispatch('core/editor').lockPostSaving(`undefined-lock-${blockClientId}-${attributeKey}`);
+};
+
+/**
+ * Given a block's client ID and an attribute key, unlocks post saving in Gutenberg.
+ * A lock is created under the name undefined-lock-[blockClientId]-[attributeKey],
+ * meaning that a lock can be created for each attribute key of each block.
+ * Multiple locks can be present and all of them have to be removed before saving is unlocked.
+ *
+ * @param {string} blockClientId A block's client ID
+ * @param {string} attributeKey The attribute key
+ */
+
+export const unlockPostEditing = (blockClientId, attributeKey) => {
+  wp.data.dispatch('core/editor').unlockPostSaving(`undefined-lock-${blockClientId}-${attributeKey}`);
+};
+
+/**
+ * Given a block's client ID, attribute key and new value, creates a lock, locks post saving in Gutenberg
+ * if the value is undefined, null or an empty string and unlocks that attribute's lock otherwise.
+ * 
+ * A lock is created under the name undefined-lock-[blockClientId]-[attributeKey],
+ * meaning that a lock can be created for each attribute key of each block.
+ * Multiple locks can be present and all of them have to be removed before saving is unlocked.
+ *
+ * @param {string} blockClientId A block's client ID
+ * @param {string} attributeKey The attribute key
+ */
+
+export const lockIfUndefined = (blockClientId, attributeKey, value) => {
+  if (typeof value === 'undefined' || value === null || value === '') {
```

Comment:
Maybe we shouldn't equate `undefined` and `null` here

Reply:
The idea was to lock post saving if there's an attribute that's missing, and I'd wager `null` and an empty string fall under that. I'm not sure what you're suggesting, but I'd suggest using the `lockPostEditing` helper for different use cases.